### PR TITLE
Added automations to verify domain in image samples using wlst and wdt

### PR DESCRIPTION
**Pull request by hzhao-github**
Added tests to verify operator Domain Home In Image samples in integration tests

All 4 tests in ItSample.java passed on Jenkins
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2814/

Passed Jenkins job:

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2849
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2850

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 1]
```diff
@@ -305,6 +304,105 @@ private void createPvPvc(String domainName) {
 
   }
 
+  private void updateDomainInputsFile(String domainName, Path sampleBase) {
+    // change namespace from default to custom, set wlst or wdt, domain name, and t3PublicAddress
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:308]: wlst or wdt is not set here

**Review by sankarpn — APPROVED** [review]
Overall looks good.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:308]: Thanks. I copy/pasted your original one :( will change

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 2]
```diff
@@ -122,86 +129,79 @@ public void testSampleDomainInPv(String model) {
...
-    final String adminServerPodName = domainName + "-" + adminServerName;
+    // update create-domain-inputs.yaml with the values from this test
+    updateDomainInputsFile(domainName, sampleBase);
 
-    final String managedServerNameBase = "managed-server";
-    String managedServerPodNamePrefix = domainName + "-" + managedServerNameBase;
-    int replicaCount = 2;
+    // update domainHomeImageBase with right values in create-domain-inputs.yaml
+    assertDoesNotThrow(() -> {
+      replaceStringInFile(Paths.get(sampleBase.toString(), "create-domain-inputs.yaml").toString(),
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:177]: I see these log messages in the test.out file which are confusing as its not actually copying the file, I think you can remove them or modify the log messages for string replacement ...https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2849/artifact/logdir/jenkins-weblogic-kubernetes-operator-kind-new-2849/wl_k8s_test_results/diagnosticlogs/ItSamples/ItSamples.out/*view*/
```
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <Copying /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <to /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <Copying /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <to /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <Copying /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <to /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <Copying /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <to /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <Copying /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <to /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <Copying /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
<10-30-2020 04:29:24> <INFO> <oracle.weblogic.kubernetes.utils.FileUtils replaceStringInFile> <to /tmp/it-results/sample-testing/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml>
```

**Review by vanajamukkara — APPROVED** [review]
looks good, please fix the log message mentioned in the comments

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:177]: It's in fiuleUtils.java, do you want me change it there?

Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:177]: yes please

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FileUtils.java [hunk 3]
```diff
@@ -278,11 +278,11 @@ public static void replaceStringInFile(String filePath, String regex, String rep
       throws IOException {
     LoggingFacade logger = getLogger();
     Path src = Paths.get(filePath);
-    logger.info("Copying {0}", src.toString());
+    logger.info("Replacing {0}", src.toString());
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FileUtils.java:281]:
```suggestion
    logger.info("Replacing {0}", regex);
```

Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FileUtils.java:281]: Oops, merged. I will remember this and change it later

**Pull request merged**